client/processors/render_processor.py

```diff
@@ -13,49 +16,128 @@
+        current_camera_id: int,
+    ):
+        for camera_id, draw in to_draw_layer:
+            if camera_id != current_camera_id:
+                current_camera_id = camera_id
+                pyray.end_mode_2d()
+                if camera_id == -1:
+                    pyray.begin_mode_2d(RenderProcessor.__default_camera)
+                else:
+                    camera = camera_by_id.get(camera_id)
+                    if camera is None:
+                        logger.warning(f"Camera with id {camera_id} not found")
+                        pyray.begin_mode_2d(RenderProcessor.__default_camera)
+                    else:
+                        pyray.begin_mode_2d(
+                            pyray.Camera2D(
+                                (camera[0].offset.x, camera[0].offset.y),
+                                (camera[1].x, camera[1].y),
+                                camera[0].rotation,
+                                camera[0].zoom,
+                            )
+                        )
+            draw()
+
+    @staticmethod
+    def __register_textures_to_draw(
+        assets_manager: AssetsManager, to_draw: map_layer_by_draw_function
+    ):
         for ent, (pos, drawable) in esper.get_components(Position, Drawable):
             texture = assets_manager.get_texture(drawable.texture_name)
             if texture is not None:
-                pyray.draw_texture(texture, int(pos.x), int(pos.y), pyray.WHITE)
-        if DEBUG_COLLIDER:
-            for ent, (pos, collider) in esper.get_components(Position, BoxCollider):
-                pyray.draw_rectangle_lines_ex(
-                    pyray.Rectangle(int(pos.x), int(pos.y), collider.x, collider.y),
-                    1,
-                    pyray.RED,
+                to_draw.setdefault(drawable.z_order, []).append(
+                    (
+                        drawable.camera_id,
+                        RenderProcessor.__create_draw_texture(texture, pos),
+                    )
                 )
+
+    @staticmethod
+    def __register_text_to_draw(
+        to_draw: map_layer_by_draw_function, r: ResourceManager
+    ):
         for ent, (pos, text) in esper.get_components(Position, Text):
-            pyray.draw_text_ex(
-                r.get_resource(AssetsManager).get_font(text.font),
-                text.value,
-                pyray.Vector2(pos.x, pos.y),
-                text.size,
-                text.spacing,
-                text.color,
+            to_draw.setdefault(text.z_order, []).append(
+                (text.camera_id, RenderProcessor.__create_draw_text_ex(r, text, pos))
             )
-        pyray.end_mode_2d()
+
+    @staticmethod
+    def __create_draw_text_ex(r, text, pos):
+        return lambda: pyray.draw_text_ex(
+            r.get_resource(AssetsManager).get_font(text.font),
+            text.value,
+            pyray.Vector2(pos.x, pos.y),
+            text.size,
+            text.spacing,
+            text.color,
+        )
```

Comment:
This is really weird, but not sure how to improve it, maybe `functools.partial`?